types/stampit/index.d.ts

```diff
@@ -1,518 +1,560 @@
+     * @param descriptors
      */
-    statics?: {};
-    /** Deeply merged properties of object instances */
-    deepProperties?: {};
-    /** Deeply merged properties of object instances */
-    deepProps?: {};
-    /** ES5 Property Descriptors applied to object instances */
-    propertyDescriptors?: {};
-    /** Deeply merged properties of Stamps */
-    staticDeepProperties?: {};
-    /** Deeply merged properties of Stamps */
-    deepStatics?: {};
-    /** ES5 Property Descriptors applied to Stamps */
-    staticPropertyDescriptors?: {};
-    /** A configuration object to be shallowly assigned to Stamps */
-    configuration?: {};
-    /** A configuration object to be shallowly assigned to Stamps */
-    conf?: {};
-    /** A configuration object to be deeply merged to Stamps */
-    deepConfiguration?: {};
-    /** A configuration object to be deeply merged to Stamps */
-    deepConf?: {};
-    /** Callback functions to execute each time a composition occurs */
-    composers?: Composer[];
+    propertyDescriptors(...descriptors: PropertyDescriptorMap[]): S̤t̤a̤m̤p̤;
+
+    /**
+     * Apply ES5 property descriptors to a Stamp and any future Stamp it composes. Creates and returns a new stamp. **Chainable**.
+     * @param descriptors
+     */
+    staticPropertyDescriptors(...descriptors: PropertyDescriptorMap[]): S̤t̤a̤m̤p̤;
 }
 
-/** Stampit Composable for main stampit() function */
-type StampitComposable = stampit.Stamp | Descriptor | Options;
+/**
+ * A function which creates a new `Stamp`s from a list of `Composable`s.
+ * @template Obj The type of the object instance being produced by the `Stamp` or the type of the `Stamp` being created (when extending a `Stamp`.)
+ */
+type ComposeMethod = typeof stampit;
 
 /**
- * Return a factory (aka Stamp) function that will produce new objects using the
- * prototypes that are passed in or composed.
- * @param options Stampit options object containing refs, methods,
- * init, props, statics, configurations, and property descriptors.
+ * A function which creates a new `Stamp`s from a list of `Composable`s.
+ * @template Obj The type of the object instance being created by the `Stamp` or the type of the `Stamp` being created (when extending a `Stamp`.)
  */
-declare function stampit(...composables: StampitComposable[]): stampit.Stamp;
+// tslint:disable-next-line: no-unnecessary-generics
+declare function stampit<Obj = any>(...composables: stampit.Composable[]): StampType<Obj>;
 
 declare namespace stampit {
+    /** A composable object (either a `Stamp` or a `Descriptor`.) */
+    type Composable = StampSignature | Descriptor<any, StampSignature>;
+
+    /**
+     * A `Stamp`'s metadata.
+     * @template Obj The type of the object instance being produced by the `Stamp`.
+     * @template S̤t̤a̤m̤p̤ The type of the `Stamp` (when extending a `Stamp`.)
+     */
+    interface Descriptor<Obj, S̤t̤a̤m̤p̤ extends StampSignature = Stamp<Obj>> {
```

Comment:
This should be renamed to "ExtendedDescriptor". The actual Descriptor do not have shortcuts like "props", "init".